astropy/coordinates/angles.py

```diff
@@ -85,54 +85,43 @@ class Angle(u.Quantity):
     _include_easy_conversion_members = True
 
     def __new__(cls, angle, unit=None, dtype=None, copy=True):
-        unit = cls._convert_unit_to_angle_unit(unit)
-        if (unit is not None and not unit.is_equivalent(u.radian)):
-            raise u.UnitsError("Requested unit {0} is not convertible to an "
-                               "angle".format(unit))
 
-        if isinstance(angle, u.Quantity):
-            # This includes Angle subclasses as well
+        if not isinstance(angle, u.Quantity):
             if unit is not None:
-                angle = angle.to(unit).value
-            else:
-                unit = angle.unit
                 unit = cls._convert_unit_to_angle_unit(unit)
-                if not unit.is_equivalent(u.radian):
-                    raise u.UnitsError(
```

Comment:
I think @taldcroft is suggesting you add tests for the *message*.  But I'm not sure I agree - in my head the type of the exception is part of the API, but the message itself is not (it's more like "documentation").  So I think it's fine as-is given that the type is checked.